### PR TITLE
Enforce passing an explicit status code in ApiResponse

### PR DESCRIPTION
**Pull request by apfelbox**

### CODE REVIEW

File: src/Api/ApiResponse.php [hunk 1]
```diff
@@ -36,4 +52,13 @@ public function withError (?string $error) : self
...
 		return $this;
 	}
+
+
+	/**
+	 *
+	 */
+	public function isOk () : bool
+	{
+		return $this->statusCode >= 200 && $this->statusCode <= 299;
```
Comment (keichinger) [inline src/Api/ApiResponse.php:62]: I guess we don't want to use the `ApiResponse` for something like a 304, this is fine.

**Review by keichinger — APPROVED** [review]

File: src/Api/ApiResponse.php (hunk 1 above)
Comment (apfelbox) [reply src/Api/ApiResponse.php:62]: Using it for a `3xx` response makes no sense, as a redirect response needs a `Location` header and may not have a body.

Comment (keichinger) [reply src/Api/ApiResponse.php:62]: For a regular browser response yes, but not in a typical REST API.

Comment (apfelbox) [reply src/Api/ApiResponse.php:62]: Even a REST API wouldn't use the payload / I would consider that a code smell.
What should it do with all the payloads it collects while following the redirects?

**Pull request merged**